readthedocs/doc_builder/director.py

```diff
@@ -347,21 +285,26 @@ def build_epub(self):
         if "epub" not in self.data.config.formats or self.data.version.type == EXTERNAL:
             return False
 
-        commands = []  # self.data.config.build.jobs.build.epub
-        if commands:
-            for command in commands:
-                self.build_environment.run(command)
-            return True
-
         # Mkdocs has no epub generation currently.
         if self.is_type_sphinx():
             return self.build_docs_class("sphinx_epub")
         return False
 
-    def post_build(self):
-        commands = []  # self.data.config.build.jobs.post_build
+    def run_build_job(self, job):
+        if (
+            getattr(self.data.config.build, "jobs", None) is None
+            or getattr(self.data.config.build.jobs, job, None) is None
+        ):
+            return
+
+        cwd = self.data.project.checkout_path(self.data.version.slug)
+        environment = self.vcs_environment
+        if job not in ("pre_checkout", "post_checkout"):
+            environment = self.build_environment
+
+        commands = getattr(self.data.config.build.jobs, job, [])
```

Comment:
Can we use `.get()` instead of `getattr()`? Is it not a dict?

Reply:
No, it's not a dict. It's a `BuildJobs` instance. The config itself is an object and each of its attributes are objects 🤷🏼

Reply:
all jobs should be already defined by this step, you can omit the `getattr` step.

Reply:
@stsewd hrm, but how do you access the field name stored in the variable `job`?

Reply:
ah, job  is a variable, then the default from `[]` could be removed, that way we can also catch errors if we pass a wrong job name.